p2p/transport/websocket/websocket_test.go

```diff
@@ -548,3 +549,64 @@ func TestResolveMultiaddr(t *testing.T) {
 		})
 	}
 }
+
+func TestListenerResusePort(t *testing.T) {
+	laddr := ma.StringCast("/ip4/127.0.0.1/tcp/5002/ws")
+	var wg sync.WaitGroup
+	var opts []Option
+	opts = append(opts, EnableReuseport())
+	_, u := newUpgrader(t)
+	tpt, err := New(u, &network.NullResourceManager{}, opts...)
+	require.NoError(t, err)
+	for i := 0; i < 2; i++ {
+		wg.Add(1)
+		go func(index int) {
+			l, err := tpt.maListen(laddr)
+			if err != nil {
+				fmt.Println("Failed to listen on websocket due to error ", err)
+			}
+			require.NoError(t, err)
+			require.Equal(t, lastComponent(t, l.Multiaddr()), wsComponent)
+			defer l.Close()
+			//Looping 4 times to ensure all 4 connections are handled.
+			for j := 0; j < 4; j++ {
+				conn, err := l.Accept()
+				if err != nil {
+					fmt.Println("Routine-", index, " Failed accepting connection due to error ", err)
+				}
+				require.NoError(t, err)
+				defer conn.Close()
+				buf := make([]byte, 6)
+				n, err := conn.Read(buf)
+				if n != 6 {
+					t.Errorf("read %d bytes, expected 2", n)
+				}
+				require.NoError(t, err)
+			}
+		}(i)
+	}
+	time.Sleep(2 * time.Second)
```

Comment:
Can we test this without sleeping for a long time?

Reply:
Took a shortcut here :). Will update the test script without sleep.